src/main/java/bio/terra/service/filedata/flight/ingest/IngestBuildAndWriteScratchLoadFileStep.java

```diff
@@ -51,11 +53,18 @@ public StepResult doSkippableStep(FlightContext context) {
               .filter(fileResultModel -> fileResultModel.getState() == BulkLoadFileState.SUCCEEDED)
               .collect(
                   Collectors.toMap(
-                      fileResultModel ->
-                          Objects.hash(
-                              fileResultModel.getSourcePath(), fileResultModel.getTargetPath()),
+                      fileResultModel -> Objects.hash(fileResultModel.getTargetPath()),
                       BulkLoadFileResultModel::getFileId));
 
+      // Part 1.1 -> Add in the already-ingested files, if told to do so
+      if (ingestRequest.isResolveExistingFiles()) {
+        Set<FileModel> existingFiles =
+            workingMap.get(IngestMapKeys.COMBINED_EXISTING_FILES, Set.class);
```

Comment:
Now that `WorkingMap` supports `TypeReference`, you can avoid the Unchecked assignment warning by writing:
```suggestion
            workingMap.get(IngestMapKeys.COMBINED_EXISTING_FILES, new TypeReference<>() {});
```
This version is also nice in that you don't have to put the type in two different places.

Reply:
Ooooooo very nice!